hazelcast-jet-core/src/main/java/com/hazelcast/jet/pipeline/Sinks.java

```diff
@@ -675,7 +681,57 @@ private Sinks() {
      * @param <T> type of the items the sink accepts
      */
     @Nonnull
-    public static <T> JmsSinkBuilder<T> jmsTopicBuilder(DistributedSupplier<ConnectionFactory> factorySupplier) {
-        return new  JmsSinkBuilder<>(factorySupplier, true);
+    public static <T> JmsSinkBuilder<T> jmsTopicBuilder(@Nonnull DistributedSupplier<ConnectionFactory> factorySupplier) {
+        return new JmsSinkBuilder<>(factorySupplier, true);
+    }
+
+    /**
+     * Returns a sink that connects to the specified database using the given
+     * {@code connectionSupplier} and writes to it using the given {@code
+     * statementFn} and {@code updateFn}. After a batch of items is written,
+     * {@code flushFn} is called to flush the changes.
+     * <p>
+     * No state is saved to snapshot for this sink. After the job is restarted,
+     * the items will likely be duplicated, providing an <i>at-least-once</i>
+     * guarantee.
+     * <p>
+     * The default local parallelism for this sink is 1.
+     * @param connectionSupplier the supplier of database connection
+     * @param statementFn the function to prepare the statement
+     * @param updateFn the function to set the parameters of the statement for
+     *                 each item received and execute the update query. Instead
+     *                 of updating for each item, it can be used to add to batch
+     *                 via {@link PreparedStatement#addBatch()}
+     * @param flushFn the function to flush the changes. Gets {@code connection}
+     *                and {@code statement} as arguments. It can be used to
+     *                commit the connection if it is not {@code autoCommit} mode
+     *                or execute the batch
+     * @param <T> type of the items the sink accepts
+     */
+    @Nonnull
+    public static <T> Sink<T> jdbc(
+            @Nonnull DistributedSupplier<Connection> connectionSupplier,
+            @Nonnull DistributedFunction<Connection, PreparedStatement> statementFn,
+            @Nonnull DistributedBiConsumer<PreparedStatement, T> updateFn,
+            @Nonnull DistributedBiConsumer<Connection, PreparedStatement> flushFn
+    ) {
+        return Sinks.fromProcessor("jdbcSink",
+                SinkProcessors.writeJdbcP(connectionSupplier, statementFn, updateFn, flushFn));
+    }
+
+    /**
+     * Convenience for {@link Sinks#jdbc(DistributedSupplier,
+     * DistributedFunction, DistributedBiConsumer, DistributedBiConsumer)}.
+     */
+    @Nonnull
+    public static <T> Sink<T> jdbc(@Nonnull String connectionUrl,
+                                   @Nonnull String updateQuery,
+                                   @Nonnull DistributedBiConsumer<PreparedStatement, T> updateFn) {
```

Comment:
It would be better if this is just `bindFn`: a function expected to bind parameters in the statement, but not expected to call `addBatch`. The `flushFn` does `executeBatch`, i'm not sure if `executeBatch` wouldn't fail if nothing was added, for example if the user used `executeUpdate` instead of `addBatch`.

Reply:
this is the convenience method, for the full version user may choose to execute the query directly in the `updateFn` rather than add to batch. I know batching is better for performance wise but this brings flexibility to the user. If you want batch, use `addBatch` and `executeBatch`. If you don't want batch just execute the query and don't do anything on flush function.

Reply:
Is there any scenario we would _not_ want to batch?

Reply:
from javadoc `SQLException if a database access error occurs, this method is called on a closed <code>Statement</code> or the driver does not support batch statements.` 

https://stackoverflow.com/questions/27079070/jdbc-driver-doesnt-support-batch-update-with-retrieval-of-identity-column-why

https://community.cloudera.com/t5/Interactive-Short-cycle-SQL/impala-jdbc-not-support-UPSERT-preparestatement-KUDU/m-p/54419

Reply:
We can still have `bindFn` expected only to bind parameters. We can query the driver if batch is supported. If it is, we'll do `addBatch/executeBatch`. If it's not, we'll do `executeQuery` only. I think it makes sense for this simple version.

Reply:
ok, we can do that